core/README.md

```diff
@@ -44,19 +44,49 @@ The `@ionic/core` package can be used in simple HTML, or by vanilla JavaScript w
 
 In addition to the default, self lazy-loading components built by Stencil, this package also comes with each component exported as a stand-alone custom element within `@ionic/core/components`. Each component extends `HTMLElement`, and does not lazy-load itself. Instead, this package is useful for projects already using a bundler such as Webpack or Rollup. While all components are available to be imported, the custom elements build also ensures bundlers only import what's used, and tree-shakes any unused components.
 
-Below is an example of importing `ion-toggle`, and initializing Ionic so it's able to correctly load the "mode", such as Material Design or iOS. Additionally, the `initialize({...})` function can receive the Ionic config.
+Below is an example of importing `ion-badge`, and initializing Ionic so it is able to correctly load the "mode", such as Material Design or iOS. Additionally, the `initialize({...})` function can receive the Ionic config.
 
 ```typescript
-import { IonBadge } from "@ionic/core/components/ion-badge";
+import { defineCustomElement } from "@ionic/core/components/ion-badge.js";
 import { initialize } from "@ionic/core/components";
 
 initialize();
+defineCustomElement();
+```
+
+Notice how we import from `@ionic/core/components` as opposed to `@ionic/core`. This helps bundlers pull in only the code that is needed.
+
+In the example above, you could also import `IonBadge` and call `customElements.define('ion-badge', IonBadge)`, but we recommend importing the `defineCustomElement` function and calling that instead. This `defineCustomElement` function will automatically define the component as well as any child components that may be required.
+
+For example, if you wanted to use `ion-modal`, you would do the following:
 
-customElements.define("ion-badge", IonBadge);
+```typescript
+import { defineCustomElement } from "@ionic/core/components/ion-modal.js";
+import { initialize } from "@ionic/core/components";
+
+initialize();
+defineCustomElement();
 ```
 
-Notice how `IonBadge` is imported from `@ionic/core/components/ion-badge` rather than just `@ionic/core/components`. Additionally, the `initialize` function is imported from `@ionic/core/components` rather than `@ionic/core`. All of this helps to ensure bundlers do not pull in more code than is needed.
+The `defineCustomElement` function will define `ion-modal`, but it will also define `ion-backdrop`, which is a component that `ion-modal` uses internally.
```

Comment:
Is this still the case? Looks like that piece was removed from overlays.ts, but maybe I'm missing how it's being done now.

Reply:
Yes. When you use an overlay controller in Vue/React, we automatically call `defineCustomElement` for the overlay you are trying to use. We do not need to do this for Angular as that integration still uses the lazy loaded version of Ionic.

If you are using @ionic/core directly, you will need to import and call `defineCustomElement` yourself before using the controller. I mention this a bit further down in the `core/README.md` file.

Reply:
Ah sorry, I'm wondering about the piece about `ion-backdrop`. Are children components also still handled? I know Sean had to add those manually to the `define` calls to fix some reference errors.

Reply:
Yeah - The `defineCustomElement` function for `ion-modal` will define `ion-backdrop`. Here is what the `defineCustomElement` function for `ion-modal` looks like:

```js
function defineCustomElement$3() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["ion-modal", "ion-backdrop"];
  components.forEach(tagName => { switch (tagName) {
    case "ion-modal":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, Modal);
      }
      break;
    case "ion-backdrop":
      if (!customElements.get(tagName)) {
        defineCustomElement$9();
      }
      break;
  } });
}

```

Reply:
Aha, got it. Makes sense 👍